misk-testing/src/main/kotlin/misk/testing/ExternalDependency.kt

```diff
@@ -0,0 +1,65 @@
+package misk.testing
+
+import io.netty.util.internal.ConcurrentSet
+import misk.logging.getLogger
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * An external dependency of the Misk Application that needs to be started for test, like Redis,
+ * Vitess, MySQL, SQS, etc.
+ */
+interface ExternalDependency {
+  /**
+   * Starts the dependency.
+   */
+  fun startup()
```

Comment:
it might be nice to have some kind of `beforeTest` hook on each external dependency. something like `TruncateTables` for a mysql/vitess external dependency

Reply:
Refactored this so that ExternalDependency is always JVM scoped. I don't think we'll actually have any that are supposed to have a test class or test method lifecycle. Makes the API simpler and now it's an interface and has `beforeEach()` and `afterEach()` methods. I'm using `afterEach()` to delete queues after every test run